showcase/src/app/shared/example-viewer/example-viewer.component.spec.ts

```diff
@@ -1,80 +1,147 @@
-import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
-import { async, ComponentFixtureAutoDetect, ComponentFixture, TestBed } from "@angular/core/testing";
+import { NO_ERRORS_SCHEMA } from "@angular/core";
+import { NoopAnimationsModule } from "@angular/platform-browser/animations";
+import { async, ComponentFixture, TestBed } from "@angular/core/testing";
 import { MatButtonModule, MatTabsModule, MatTooltipModule } from "@angular/material";
+import { throwError, of, Observable, Subject } from "rxjs";
+import { filter, delay } from "rxjs/operators";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { StarkPrettyPrintModule } from "@nationalbankbelgium/stark-ui";
 
-import { ExampleViewerComponent } from "./example-viewer.component";
+import { ExampleFile, ExampleViewerComponent } from "./example-viewer.component";
 import { FileService } from "./file.service";
-import { throwError, of } from "rxjs";
-import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
-import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
 import SpyObj = jasmine.SpyObj;
-import { StarkPrettyPrintModule } from "@nationalbankbelgium/stark-ui";
-import { NO_ERRORS_SCHEMA } from "@angular/core";
+import Spy = jasmine.Spy;
 
 describe("ExampleViewerComponent", () => {
 	let component: ExampleViewerComponent;
-	let fileService: FileService;
+	let fileService: SpyObj<FileService>;
 	let fixture: ComponentFixture<ExampleViewerComponent>;
 	let logger: SpyObj<StarkLoggingService>;
 
 	beforeEach(async(() => {
 		return TestBed.configureTestingModule({
 			declarations: [ExampleViewerComponent],
-			imports: [BrowserAnimationsModule, HttpClientModule, MatButtonModule, MatTabsModule, MatTooltipModule, StarkPrettyPrintModule],
+			imports: [NoopAnimationsModule, MatButtonModule, MatTabsModule, MatTooltipModule, StarkPrettyPrintModule],
 			providers: [
-				{ provide: ComponentFixtureAutoDetect, useValue: true },
 				{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() },
-				FileService
+				{
+					provide: FileService,
+					useValue: jasmine.createSpyObj("FileServiceSpy", ["fetchFile"])
+				}
 			],
 			schemas: [NO_ERRORS_SCHEMA] // tells the Angular compiler to ignore unrecognized elements and attributes: mat-icon
 		}).compileComponents();
 	}));
 
 	beforeEach(() => {
 		logger = TestBed.get(STARK_LOGGING_SERVICE);
+		fileService = TestBed.get(FileService);
+		fileService.fetchFile.and.callFake(() => of("initial dummy file content"));
+
 		fixture = TestBed.createComponent(ExampleViewerComponent);
 		component = fixture.componentInstance;
-		fileService = fixture.debugElement.injector.get<FileService>(FileService as any);
+		fixture.detectChanges(); // trigger initial data binding
```

Comment:
Don't you think it will be easier to use ComponentFixtureAutoDetect

https://angular.io/guide/testing#automatic-change-detection

Reply:
I thought so, but then I realized it is not working in all cases. In the link you mention it is stated the following:

> The second and third test reveal an important limitation. The Angular testing environment does not know that the test changed the component's title. The ComponentFixtureAutoDetect service responds to asynchronous activities such as promise resolution, timers, and DOM events. But a direct, synchronous update of the component property is invisible. The test must call fixture.detectChanges() manually to trigger another cycle of change detection.

> Rather than wonder when the test fixture will or won't perform change detection, the samples in this guide always call detectChanges() explicitly. There is no harm in calling detectChanges() more often than is strictly necessary.

And to be honest I have the same feeling, I don't like the fact that it will work in some cases and it won't in some others (like in the line 45 of the old code in this test file).
Therefore I prefer to explicitly trigger change detection and know what is going on at all times and avoid some magic done by Angular in "some cases".